include/lmms_math.h

```diff
@@ -30,27 +30,27 @@
 #include <cmath>
 #include <cstdint>
 
+#include "lmms_basics.h"
 #include "lmms_constants.h"
 #include "lmmsconfig.h"
 #include <cassert>
 
 namespace lmms
 {
-
-#ifdef __INTEL_COMPILER
-
-static inline float absFraction( const float _x )
-{
-	return( _x - floorf( _x ) );
-}
-
-static inline float fraction( const float _x )
+/*!
+ * @brief Returns the fractional part of a float, a value between -1.0f and 1.0f.
+ *
+ * fraction( 2.3) =>  0.3
+ * fraction(-2.3) => -0.3
+ *
+ * Note that if the return value is used as a phase of an oscillator, that the oscillator must support
+ * negative phases.
+ */
+static inline float fraction(const float x)
```

Comment:
> Function parameters passed by value are rarely mutated, but also rarely declared `const`. To avoid confusion and lots of false positives, don’t enforce this rule for function parameters.

I've seen this misunderstood before. Note the use of "enforce", not "follow": this is not a recommendation to avoid `const` function parameters, but a recommendation for diagnostic tool authors not to emit warnings for non-`const` function parameters. There's nothing wrong with declaring function parameters `const`, and in fact it's a good idea in principle.

Reply:
I agree @DomClark, and I probably mistakenly understood that as saying "follow" instead of "enforce". It's fine to use `const` here, I just find it somewhat odd for value parameters, but it is good when you clearly want to express that a parameter will not be changed when passed into the function. @Rossmaxx, feel free to revert this suggestion, I will give you the choice.

Reply:
I'll add const back